src/rule_evaluator.ts

```diff
@@ -40,24 +55,60 @@ function evaluateCondition(subjectAttributes: Record<string, any>, condition: Co
       case OperatorType.MATCHES:
         return new RegExp(condition.value as string).test(value as string);
       case OperatorType.ONE_OF:
-        return isOneOf(value, condition.value);
+        return isOneOf(
+          value.toString().toLowerCase(),
+          condition.value.map((value: string) => value.toLowerCase()),
+        );
       case OperatorType.NOT_ONE_OF:
-        return isNotOneOf(value, condition.value);
+        return isNotOneOf(
+          value.toString().toLowerCase(),
+          condition.value.map((value: string) => value.toLowerCase()),
+        );
+    }
+  }
+  return false;
+}
+
+function evaluateObfuscatedCondition(
```

Comment:
Separate function for evaluating conditions for the obfuscated RAC. We decode values in the RAC if we know base64 encoding was used and do the evaluation, otherwise we hash the arguments to getAssignment to do string matching